releases/release-5.1.2.md

```diff
@@ -0,0 +1,152 @@
+    - This reverts commit d4cd12fe422fb18b6012607ee18b6acca40d9225. [#28091](https://github.com/pingcap/tidb/pull/28091)
+    - planner: fix the problem of using `enum like 'x%'` to build the wrong range [#28066](https://github.com/pingcap/tidb/pull/28066)
+    - fix expression rewrite makes between expr infers wrong collation. [#27549](https://github.com/pingcap/tidb/pull/27549)
+    - execution: fix cte dead lock when used with IndexLookupJoin [#27536](https://github.com/pingcap/tidb/pull/27536)
+    - Fix a bug that retryable deadlocks are incorrectly recorded into `INFORMATION_SCHEMA.DEADLOCKS` table. [#27535](https://github.com/pingcap/tidb/pull/27535)
+    - make `group_concat` function consider the collation [#27529](https://github.com/pingcap/tidb/pull/27529)
+    - Fix bug that count disctinct on multi-columns return wrong result when new collation is on. [#27506](https://github.com/pingcap/tidb/pull/27506)
+    - Fix an issue that the `TABLESAMPLE` query result from partitioned tables is not sorted as expected. [#27411](https://github.com/pingcap/tidb/pull/27411)
+    - expression: fix extract bug when argument is a negative duration [#27367](https://github.com/pingcap/tidb/pull/27367)
+    - The undocumented `/debug/sub-optimal-plan`  HTTP API has been removed. [#27265](https://github.com/pingcap/tidb/pull/27265)
+    - fix wrong selection push down when having above agg [#27258](https://github.com/pingcap/tidb/pull/27258)
+    - fix a bug that query on hash partition table return wrong reslut [#27164](https://github.com/pingcap/tidb/pull/27164)
+    - executor: fix unexpected behavior when casting invalid string to date [#27112](https://github.com/pingcap/tidb/pull/27112)
+    - fix a bug that creates partition fail if `NO_UNSIGNED_SUBTRACTION` is set. [#27053](https://github.com/pingcap/tidb/pull/27053)
+    - planner: add missing distinct flag for Apply convert to join. [#26969](https://github.com/pingcap/tidb/pull/26969)
+    - Fix an issue that NO_ZERO_IN_DATE does not work on the default values. [#26904](https://github.com/pingcap/tidb/pull/26904)
+    - store/copr: block the tiflash node for a period when it fails before. [#26897](https://github.com/pingcap/tidb/pull/26897)
+    - Fix then bug if the CTE is referenced more than once. [#26661](https://github.com/pingcap/tidb/pull/26661)
+    - planner: fix CTE bug when MergeJoin is used [#26658](https://github.com/pingcap/tidb/pull/26658)
+    - Fix a bug 'select for update' does not correctly lock the data when a normal table join a partition table. [#26631](https://github.com/pingcap/tidb/pull/26631)
+    - Fix a panic when select for update on a normal table join a partition table [#26563](https://github.com/pingcap/tidb/pull/26563)
+    - Fix the issue that point get does not use lite version resolve lock [#26562](https://github.com/pingcap/tidb/pull/26562)
```

Comment:
```suggestion
    -  Set a block duration for the newly recovered TiFlash node to avoid blocking queries during this time [#26897](https://github.com/pingcap/tidb/pull/26897)
    - Fix a bug that might occur when the CTE is referenced more than once [#26661](https://github.com/pingcap/tidb/pull/26661)
    - Fix a CTE bug when MergeJoin is used [#26658](https://github.com/pingcap/tidb/pull/26658)
    - Fix a bug that the 'SELECT FOR UPDATE' statement does not correctly lock the data when a normal table joins a partition table [#26631](https://github.com/pingcap/tidb/pull/26631)
    - Fix the issue that the 'SELECT FOR UPDATE' statement returns an error when a normal table joins a partition table [#26563](https://github.com/pingcap/tidb/pull/26563)
    - Fix the issue that `PointGet` does not use the lite version of resolve lock [#26562](https://github.com/pingcap/tidb/pull/26562)
```